pyi.py

```diff
@@ -346,6 +346,12 @@ def _is_object(node: ast.expr | None, name: str, *, from_: Container[str]) -> bo
     )
 
 
+def _is_builtins_object(node: ast.expr | None) -> bool:
+    return _is_object(node, name="object", from_={"builtins"}) or _is_object(
+        node, name="Unused", from_={"_typeshed"}
+    )
```

Comment:
I think I'd rather keep the `_is_builtins_object` as it is, so that it does what it says on the tin (tells you whether the node represents `builtins.object` or not). This patch would change the function so that it tells you whether the node represents `builtins.object` or this other symbol that's an alias for `builtins.object` or not. That's slightly different (`_typeshed.Unused` has a different semantic meaning to the human reader, even if it's all the same to a type checker), and might impact our ability to reuse the `_is_builtins_object` function in the future, if we ever add more checks that to test whether a node represents `builtins.object` or not.

How about we do something like this, instead?

```python
_is_builtins_object = partial(_is_object, name="object", from_={"builtins"})
_is_Unused = partial(_is_object, name="Unused", from_={"_typeshed"})

def _is_object_or_Unused(node: ast.expr | None) -> bool:
    return _is_builtins_object(node) or _is_Unused(node)
```

And then we use `_is_object_or_Unused` in Y036 instead of `_is_builtins_object`?